zuul-core/src/main/java/com/netflix/zuul/message/http/HttpRequestMessageImpl.java

```diff
@@ -123,6 +126,7 @@ public HttpRequestMessageImpl(SessionContext context, String protocol, String me
         this.scheme = scheme;
         this.port = port;
         this.serverName = serverName;
+        this.clientRemoteAddress = clientRemoteAddress;
```

Comment:
checkNotNull()

Reply:
Not sure we'd want to enforce this knowing the attribute might be absent.

Reply:
If it's really possible to be absent, then it needs to be marked `@Nullable`